src/module_2/module_2_1.md

```diff
@@ -0,0 +1,402 @@
+                                      df_clean["count_children"].isna() |
+                                      df_clean["count_babies"].isna() |
+                                      df_clean["count_pets"].isna()),
+                                      1, 0)
+
+missing_columns = [
+    "count_adults",
+    "count_children",
+    "count_babies",
+    "count_pets",
+]
+
+for column in missing_columns:
+    df_clean[column].fillna(df_clean[column].mean(), inplace=True)
+
+df_clean.to_parquet(f"{folder_path}/users_clean.parquet")
+
+```
+
+#### Dejo comentado lo que hice mal: Sustituir missing values con random values siguiendo la distribucion de cada variable 
+
+
+```python
+
+
+# 95% of rows with missing info about people
+# Comparing df.isna() with df.notna(), similar user_segment and user_nuts1 distribution
+# Fill every missing value following column distribution
+
+# missing_columns = [
+#     "count_adults",
+#     "count_children",
+#     "count_babies",
+#     "count_pets",
+#     "user_nuts1",
+# ]
+
+# for column in missing_columns:
+#     missing_values = df_clean[column].isna()
+#     s = df_clean[column].value_counts(normalize=True)
+#     np.random.seed(99)
+#     df_clean.loc[missing_values, column] = np.random.choice(
+#         s.index, size=missing_values.sum(), p=s.values
+#     )
+
+# # Recalculating count_people column where np.nan
+# missing_values = df_clean["count_people"].isna()
+# df_clean.loc[missing_values, "count_people"] = (
+#     df_clean.loc[missing_values, "count_adults"]
+#     + df_clean.loc[missing_values, "count_children"]
+#     + df_clean.loc[missing_values, "count_babies"]
+# )
+
+# # Exporting dataframe to clean folder
+# df_clean.to_parquet(f"data-modified/users.parquet")
+```
+
+
+```python
+# Clean Inventory dataframe
+# Importing dataframe 
+df = pd.read_parquet(f"{folder_path}/inventory.parquet")
+
+#Adding np.nan to products with price equal to zero
+df["price"].mask(df["price"] == 0, np.nan, inplace=True)
+df["compare_at_price"].mask(df["compare_at_price"] == 0, np.nan, inplace=True)
+df.to_parquet(f"{folder_path}/inventory_clean.parquet")
+
+```
+
+#### Añadiendo informacion de usuarios a cada pedido
+
+
+```python
+# Merging orders dataframe
+
+df_orders = pd.read_parquet(f"{folder_path}/orders_clean.parquet")
+df_users = pd.read_parquet(f"{folder_path}/users.parquet")
+df_inventory = pd.read_parquet(f"{folder_path}/inventory_clean.parquet")
+
+df_final = df_orders.copy()
+
+
+# Left join to complete users information (count adults, count children, etc.)
+df_final = pd.merge(df_final, df_users, how="left", on=["user_id"])
+
+
+# Explode list of items whitin orders
+df_explode = df_final.explode("ordered_items")
+df_explode.rename(columns={"ordered_items": "variant_id"}, inplace=True)
+
+# Merging with inventory dataframe
+df_explode = pd.merge(df_explode, df_inventory, how="left", on=["variant_id"])
+df_explode.dropna(inplace=True)
```

Comment:
This will drop every row with at least one nan. Why do you do a `left` join rather than a `inner` join if you are going to do so? 

I think it's too risky without checking first where do you have the nans, and how many have been created by the left join.